File: app/src/main/java/com/waygo/data/model/butler/ButlerShowResponse.java
package com.waygo.data.model.butler;

import android.graphics.Bitmap;
import android.support.annotation.NonNull;

import com.waygo.utilskt.Result;

public class ButlerShowResponse extends ButlerResponse {

    @NonNull
    private final Bitmap mImage;

    private ButlerShowResponse(@NonNull final String message,
                              @NonNull final Bitmap image) {
        super(message);
        mImage = image;
    }

    @NonNull
    public static Result<ButlerResponse> create(@NonNull final String message,
                                                 @NonNull final Bitmap image) {
        return createMessage(message)
                .map(m -> (ButlerResponse)new ButlerShowResponse(message, image));
    }

    @NonNull
    public Bitmap getImage() {
        return mImage;
    }
}

File: app/src/main/java/com/waygo/data/model/butler/ButlerShowResponse.kt
package com.waygo.data.model.butler

import android.graphics.Bitmap

import com.waygo.utilskt.Result
import kotlin.platform.platformStatic

public class ButlerShowResponse private constructor(message: String, public val image: Bitmap) : ButlerResponse(message) {
    companion object {
        public platformStatic fun create(message: String, image: Bitmap): Result<ButlerShowResponse> =
                ButlerResponse.createMessage(message)
                        .map { ButlerShowResponse(it, image) }
    }
}
